server/service/osquery.go

```diff
@@ -1795,6 +1797,17 @@ func (svc *Service) processSoftwareForNewlyFailingPolicies(
 			level.Debug(logger).Log("msg", "installer platform does not match host platform")
 			continue
 		}
+		scoped, err := svc.ds.IsSoftwareInstallerLabelScoped(ctx, failingPolicyWithInstaller.InstallerID, hostID)
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "checking if software installer is label scoped to host")
+		}
+		if !scoped {
+			// NOTE: we update the policy status here to stop it from showing up as "failed" in the
+			// host details.
+			incomingPolicyResults[failingPolicyWithInstaller.ID] = nil
+			level.Debug(logger).Log("msg", "not marking policy as failed since software is out of scope for host")
+			continue
```

Comment:
Oh I think this was the missing piece for my understanding, we change the policy result to undefined instead of failed so that if the host eventually becomes a target for that installer, the policy will be detected as newly-failed, right?

Reply:
exactly @mna !